src/lib/factories.js

```diff
@@ -30,23 +30,39 @@ const mergePropsAndClassName = (defaultProps, props) => {
  * @param {function|string} Component A ReactClass or string
  * @param {function} mapValueToProps A function that maps a primitive value to the Component props
  * @param {string|object|function} val The value to create a ReactElement from
- * @param {object} defaultProps Default props to add to the final ReactElement
+ * @param {object|function} [defaultProps] Default props object or function (called with regular props).
```

Comment:
I struggled with that, suggestions welcome.  It is one of 3 props.  The props from the element the user passed, a props object the user passed, or the props generated by the factory's `mapValueToProps` using string/number literal the user passed.